# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/LIROps.yaml
# Commit: 8129ee5398d5
# Full Hash: 8129ee5398d5ce34183dd2ca51c3aa1d27e9c318
# Author: Doug Thayer <dothayer@mozilla.com>
# Date: 2023-01-20 03:46:23
# Regressor Bug: 1809359
# File Overlap Count: 1
# Description:
#   Bug 1809359 - Inline SetElementMegamorphic r=jandem
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D166401
# ==============================================================================

diff -r a4156a9d0b2b -r 8129ee5398d5 js/src/jit/LIROps.yaml
--- a/js/src/jit/LIROps.yaml	Thu Jan 19 22:22:32 2023 +0000
+++ b/js/src/jit/LIROps.yaml	Thu Jan 19 23:53:40 2023 +0000
@@ -2193,6 +2193,15 @@
     object: WordSized
     index: BoxedValue
     value: BoxedValue
+  # On x86 we do not have enough registers to use 3 temps for this *and* take
+  # five words worth of operands. Since it's 32-bit, though, we get two
+  # registers from pushing `value`, which doesn't get used until the end
+  # anyway. This is somewhat klunky, but oh well.
+#ifdef JS_CODEGEN_X86
+  num_temps: 1
+#else
+  num_temps: 3
+#endif
   call_instruction: true
   mir_op: true
 